src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java

```diff
@@ -195,4 +208,56 @@ public ResponseEntity<Void> deleteCompany(@PathVariable Long id) {
         companyService.delete(id);
         return ResponseEntity.ok().build();
     }
+
+    /* ONLY ROLE_ADMIN */
+    @DeleteMapping("/companies/service/{id}")
+    public ResponseEntity<Void> deleteServiceAccount(@PathVariable Long id) {
+        Optional<CompanyDTO> companyDto = companyService.findOne(id);
+        if (companyDto.isPresent()) {
+            companyService.deleteServiceAccount(companyDto.get());
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok().build();
+    }
+
+    /* ONLY ROLE_ADMIN */
+    @GetMapping("/companies/service/token/{id}")
+    public ResponseEntity<List<Token>> getServiceAccountTokensForCompany(@PathVariable Long id) {
+        return ResponseUtil.wrapOrNotFound(companyService.getServiceAccountTokensForCompany(id));
+    }
+
+    /* ONLY ROLE_COMPANY_ADMIN */
+    @PostMapping("/companies/service/token") 
```

Comment:
I choose not to use the ID here because a company admin should only be able to create a service account token fro their own company. Thus, specifying the company id is unnecessary as we should always get the company Id by the current user.